assemblies/core/static/src/main/resources-filtered/spoon.sh

```diff
@@ -233,12 +233,17 @@ export LIBPATH
 # **************************************************
 # ** Setup Karaf endorsed libraries directory     **
 # **************************************************
-
-JAVA_ENDORSED_DIRS=""
-if [ ! -z "$_PENTAHO_JAVA_HOME" ]; then
-	JAVA_ENDORSED_DIRS="${_PENTAHO_JAVA_HOME}/jre/lib/endorsed:${_PENTAHO_JAVA_HOME}/lib/endorsed:"
+if $($_PENTAHO_JAVA -version 2>&1 | grep "version \"1\.8\..*" > /dev/null ) 
+then
+	JAVA_ENDORSED_DIRS=""
+	if [ ! -z "$_PENTAHO_JAVA_HOME" ]; then
+		JAVA_ENDORSED_DIRS="${_PENTAHO_JAVA_HOME}/jre/lib/endorsed:${_PENTAHO_JAVA_HOME}/lib/endorsed:"
+	fi
+	JAVA_ENDORSED_DIRS="${JAVA_ENDORSED_DIRS}${BASEDIR}/system/karaf/lib/endorsed"
+	JAVA_ENDORSED_DIRS="-Djava.endorsed.dirs="${JAVA_ENDORSED_DIRS}
+else
+    JAVA_ENDORSED_DIRS="-Djava.locale.providers=COMPAT,SPI"
```

Comment:
Same as above, probably clearer to just use a new variable name for locale.

Reply:
fixed